src/components/Microsoft.DotNet.UpgradeAssistant.Extensions/ExtensionManager.cs

```diff
@@ -143,30 +135,59 @@ private static ExtensionInstance LoadOptionsExtension(IEnumerable<AdditionalOpti
             return new ExtensionInstance(new PhysicalFileProvider(Environment.CurrentDirectory), Environment.CurrentDirectory, config);
         }
 
-        public async Task<bool> RemoveAsync(string name, CancellationToken token)
+        public Task<bool> RemoveAsync(string name, CancellationToken token)
         {
-            var config = await _configurationLoader.LoadAsync(token).ConfigureAwait(false);
+            var config = _configurationLoader.Load();
             var existing = config.Extensions.FirstOrDefault(e => string.Equals(e.Name, name, StringComparison.OrdinalIgnoreCase));
 
             if (existing is null)
             {
-                return false;
+                return Task.FromResult(false);
```

Comment:
I only see sync method calls in here , why is this needed?

Reply:
Also why is the Remove an async if none of the method calls are async ?

Reply:
To remain consistent with the other signatures. Also, it may end up being async and reduces change (for instance, one of these methods went from sync->async and it caused a bit of annoying churn)